docs/guides/getting-started/getting-started-scaling-and-tuning.adoc

```diff
@@ -0,0 +1,62 @@
+
+To limit memory utilization of queued work and to provide for load shedding see <@links.ha id="concepts-threads#load-shedding" /> 
+
+If you are experiencing timeouts in obtaining database connections, you should consider increasing the number of connections available - see <@links.ha id="concepts-database-connections" />.
+
+=== Vertical Autoscaling
+
+Some platforms, such as Kubernetes, provide mechanisms to vertically autoscale. That is not recommended for {project_name} if it requires restarting the server instance - which is currently the case for Java on Kubernetes. You can consider instead providing higher CPU and/or memory limits to your container to allow your JVM to adapt within those limits as needed.
+
+== Horizontal Scaling
+
+A single {project_name} instance is susceptible to availability issues. If the instance goes down you will be faced with a full outage, and until the persistent sessions feature is available all of your on-going sessions will be lost. By running at least two cluster members on different machines you will greatly increase the availability of {project_name}. 
+
+There will also be a limit on the amount of concurrent requests a single VM can effectively handle. Additional server instances can provide roughly linear scaling of throughput until associated resources, such as the database or distributed caching limit scaling.
+
+In general it's best to let the {project_name} Operator handle horizontal scaling concerns. When using the Operator simply set the Keycloak custom resource spec.instances as desired to horizontally scale - see <@links.ha id="deploy-keycloak-kubernetes" /> for more.
+
+If you are not using the Operator, please review the following.
+
+Your instances should be on separate machines if possible to increase availability. On Kubernetes you can use Pod anti-affinitity to enforce this.
+
+You need to use distributed or external caching for cluster members to share the same session state - see <@links.server id="caching" /> for more on the relevant configuration. The embedded Infinispan cache has horizontal scaling considerations including:
+- Your instances will need a way to find each other, see discovery in the server guide.
+- It is not well suited for clusters crossing availability zones, also known as a stretch clusters. It is recommended that when using the embedded Infinispan cache that the instances all exist in the same availability zone. On Kubernetes you can use pod affinity to enforce this grouping of pods.
+- It does not gracefully handle multiple members joining at the same time. On Kubernetes you can use a StatefulSet using the default of serial handling to ensure Pods are started individually.
```

Comment:
Multiple members joining simultaneously isn't really an issue, however > 1 member leaving the cluster simultaneously can cause data loss.

Reply:
@ryanemerson - I'll update this to mention the scale down issue. From #11763, #21108 and other issues, we've also been mentioning a parallel join problem. Is that something later versions of infinispan have addressed? And not related to infinispan, but I believe we've also seen liquibase migration problem when pods start in parallel.

Reply:
> we've also been mentioning a parallel join problem. Is that something later versions of infinispan have addressed?

I'm not aware of an issue that could lead to data loss, or blocking. There maybe some additional network overhead due to rebalances being restarted once all members are added. @pruivo any thoughts?

Reply:
José found some issues. Those are fixed but it may be more out there :) 
* https://issues.redhat.com/browse/ISPN-15432
* https://issues.redhat.com/browse/ISPN-15586
* https://issues.redhat.com/browse/ISPN-15872

Reply:
The text now mentions both leaving and joining. Does that seem appropriate, or are the join issues sufficiently addressed?

Reply:
I'm happy with the current text :+1: